Doc/reference/datamodel.rst

```diff
@@ -1455,10 +1455,13 @@ access (use of, assignment to, or deletion of ``x.name``) for class instances.
 
 .. method:: object.__getattr__(self, name)
 
-   Called when an attribute lookup has not found the attribute in the usual places
-   (i.e. it is not an instance attribute nor is it found in the class tree for
-   ``self``).  ``name`` is the attribute name. This method should return the
-   (computed) attribute value or raise an :exc:`AttributeError` exception.
+   Called when an attribute lookup has not found the attribute in the usual
+   places, namely when it is a :func:`property` attribute whose :meth:`__get__`
+   method raises an :exc:`AttributeError`, when :meth:`__getattribute__` raises
+   an :exc:`AttributeError` because *name* is not an instance attribute, or
+   when it is not found in the class tree for ``self``.  This method should
+   return the (computed) attribute value or raise an :exc:`AttributeError`
+   exception.
```

Comment:
You changed my suggested revision.  I am guessing because it was slightly off, in that when a property raises, __getattribute__ does not.  However, I prefer to change 'usual places' because a property is not a place.  I also prefer referring to properties second.  How about:

Called when normal attribute lookup fails with an AttributeError.  Either *name* is neither an instance attribute nor an attribute in the class tree for ``self``, and meth:`__getattribute__` raises an :exc:`AttributeError`; or :meth:`__get__` of a *name* property raises :exc:`AttributeError`.  This method should either return ... or raise... .

Any further revision?

Reply:
I left out the backticks needed to quote double underscores so that they are not interpreted as __bold__ markup.

Reply:
I didn't mean to take away any of the intention with your original suggested revision.  From other tickets I've looked at, I've learned that there's a balance between maintaining some of the existing conversational feel to the docs (especially if it's existed for a long time) when modifying the docs, so I was trying to incorporate that.  I was trying not to step on toes (and I stepped on yours  :-)  )

I had moved properties first because when looking at the attribute access order docs, it said that a descriptor method might override the default behavior, so I tried to indicate an order of application, but I don't think it matters.  Actually, maybe instead of 'normal attribute lookup', we should use 'default attribute access', like in https://docs.python.org/3/howto/descriptor.html?highlight=descriptors#definition-and-introduction?

For revisions, the 'Either.. neither...' line somehow doesn't sound quite right.  Maybe it should all be after a semi-colon or dash?

Called when the default attribute access fails with an AttributeError - either meth:__getattribute__ raises an :exc:AttributeError because name is not an instance attribute or an attribute in the class tree for `self`; or :meth:__get__ of a name property raises :exc:AttributeError. This method should either return ... or raise... .

Reply:
@terryjreedy   Do you think a test should be added to test_descr.py for this scenario (ie, combining a property with `__getattr__`)?